app/src/main/java/com/github/ravifrancesco/softwaredevproject/LineOfSight.java

```diff
@@ -0,0 +1,151 @@
+package com.github.ravifrancesco.softwaredevproject;
+
+import android.util.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This class enables the computation of the visible POIPoints from the user's location.
+ * It uses the ElevationMap class to compute the elevation of the terrain at a given location.
+ *
+ * The ELEVATION_DIFFERENCE_THRESHOLD represents the maximum acceptable difference in meters
+ * between the line that connects the user to the POIPoint and the actual elevation of the
+ * terrain in a given point.
+ */
+public class LineOfSight {
+
+    static final int ELEVATION_DIFFERENCE_THRESHOLD = 50; // in meters
+
+    private UserPoint userPoint;
+
+    private ElevationMap elevationMap;
```

Comment:
elevationMap can be made final
```suggestion
    private final ElevationMap elevationMap;
```